0000-open-options.md

```diff
@@ -0,0 +1,543 @@
+On Windows this behaviour is imitated by specifying not only `CREATE_NEW` but
+also `FILE_FLAG_OPEN_REPARSE_POINT`.
+
+Simply put: nothing is allowed to exist on the target location, also no
+(dangling) symlink.
+
+if `.create_new(true)` is set, `.create()` and `.truncate()` are ignored.
+
+
+### Unix-specific: Mode
+`.mode(0o666)`
+
+On Unix the new file is created by default with permissions `0o666` minus the
+systems `umask` (see [Wikipedia](https://en.wikipedia.org/wiki/Umask)). It is
+possible to set on other mode with this option.
+
+If a file already exist or `.create(true)` or `.create_new(true)` are not
+specified, `.mode()` is ignored.
+
+Rust currently does not expose a way to modify the umask.
+
+### Windows-specific: Attributes
+`.attributes(FILE_ATTRIBUTE_READONLY | FILE_ATTRIBUTE_HIDDEN | FILE_ATTRIBUTE_SYSTEM)`
+
+Files on Windows can have several attributes, most commonly one or more of the
+following four: readonly, hidden, system and archive. Most
+[others](https://msdn.microsoft.com/en-us/library/windows/desktop/gg258117%28v=vs.85%29.aspx)
+are properties set by the file system. It seems of the others only
+`FILE_ATTRIBUTE_ENCRYPTED` and `FILE_ATTRIBUTE_TEMPORARY` are meaningful to set
+when creating a new file.
+
+It is no use to set the archive flag, as Windows sets it automatically when the
+file is newly created or modified. This flag may then be used by backup
+applications as an indication of which files have changed.
+
+If a _new_ file is created because it does not yet exist and `.create(true)` or
+`.create_new(true)` are specified, the new file is given the attributes declared
+with `.attributes()`.
+
+If an _existing_ file is opened with `.create(true).truncate(true)`, its
+attributes are also replaced by the ones specified with `.attributes()`.
+
+In all other cases the attributes get ignored.
+
+
+## Sharing / locking
+On Unix it is possible for multiple processes to read and write to the same file
+at the same time.
+
+When you open a file on Windows, the system by default denies other processes to
+read or write to the file, or delete it. By setting the sharing mode, it is
+possible to allow other processes read, write and/or delete access. For
+cross-platform consistency, Rust imitates Unix by setting all sharing flags.
+
+Unix has no equivalent to the kind of file locking that Windows has. It has two
+types of advisory locking, POSIX and BSD-style. Advisory means any process that
+does not use locking itself can happily ignore the locking af another process.
+As if that is not bad enough, they both have
+[problems](http://0pointer.de/blog/projects/locking.html) that make them close
+to unusable for modern multi-treaded programs. Linux may in some very rare cases
+support mandatory file locking, but it is just as broken as advisory.
+
+For Rust, the sharing mode can be set with a Windows-specific option. Given the
+problems above, i don't expect there to ever be a cross-platform option for file
```

Comment:
In near future, perhaps not, but I’m pretty sure a way to do it would get implemented in a timely manner should a reason to do so arise.